feat: document new experimental feature `preserveScriptsOrder` (#11211)
withastro/docs: src/content/docs/en/reference/experimental-flags/preserve-scripts-order.mdx

---
title: Experimental preserve scripts order
sidebar:
  label: Preserve scripts order
i18nReady: true
---

import Since from '~/components/Since.astro'

<p>

  **Type:** `boolean`<br />
  **Default:** `false`<br />
  <Since v="5.5.0" />
</p>

Renders multiple `<style>` and `<script>` tags in the same order as they were declared in the source code.

> To enable this virtual module, add the `experimental.preserveScriptOrder` feature flag to your Astro config:

Reviewer: Just a small nit!
```suggestion
To enable this behavior, add the `experimental.preserveScriptOrder` feature flag to your Astro config:
```

```js title="astro.config.mjs" ins={4-6}
import { defineConfig } from "astro/config"

export default defineConfig({
  experimental: {
    preserveScriptOrder: true
  }
})
```

## Usage

> This experimental flag requires no specific usage, and only affects the order in which Astro renders your styles and scripts.

Reviewer 2: comma nit
```suggestion
This experimental flag requires no specific usage and only affects the order in which Astro renders your styles and scripts.
```

> When rendering multiple `<style>` and `<script>` tags on the same page, Astro currently reverses their order in your generated HTML output. This can give unexpected results, for example CSS styles being overridden by earlier defined style tags when your site is built. This experimental flag instead renders `<script>` and `<style>` tags in the order they are defined.

Reviewer 2: comma nit
```suggestion
When rendering multiple `<style>` and `<script>` tags on the same page, Astro currently reverses their order in your generated HTML output. This can give unexpected results, for example, CSS styles being overridden by earlier defined style tags when your site is built. This experimental flag instead renders `<script>` and `<style>` tags in the order they are defined.
```

For example, the following component has two `<style>` tags and two `<script>` tags:

> ```html

Reviewer 3: ```suggestion
```astro title="src/components/MyComponent.astro"
```
